packages/opentelemetry-node/src/hooks.ts

```diff
@@ -0,0 +1,69 @@
+import type { Span as OtelSpan } from '@opentelemetry/sdk-trace-base';
+import type { Span } from '@sentry/types';
+
+/**
+ * Keep some otel hooks separate from the default client hooks.
+ */
+export class OtelHooks {
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  private _hooks: Record<string, Function[]> = {};
+
+  /**
+   * Fires when an Otel Span is ended.
+   * Receives the Otel Span as argument.
+   * Return `false` from the callback to skip this span in Sentry.
+   */
+  public on(hook: 'spanEnd', callback: (otelSpan: OtelSpan, sentrySpan: Span) => void | false): void;
```

Comment:
Why can't we just use hooks on the default client? We can just append with `experimental`

I'd like to avoid using a separate class if we can.

Reply:
I changed it to use `otelSpanEnd` hook on the general client!